upload_log_files.py

```diff
@@ -0,0 +1,139 @@
+import argparse
+import os
+import re
+
+from core_data_modules.logging import Logger
+from storage.google_cloud import google_cloud_utils
+
+from src.lib import PipelineConfiguration
+
+log = Logger(__name__)
+
+
+def get_file_paths(dir_path):
+    # search for .gzip (data archive) and .profile (memory profile) files only because os.listdir(dir_path)
+    # returns all files in the directory
+    log_files_list = [file for file in os.listdir(dir_path) if file.endswith((".gzip", ".profile"))]
+    log_file_paths = [os.path.join(dir_path, basename) for basename in log_files_list]
+
+    return log_file_paths
+
+def fetch_latest_modified_file_path(dir_path):
+    log_file_paths = get_file_paths(dir_path)
+    latest_modified_log_file = max(log_file_paths, key=os.path.getmtime)
+
+    return latest_modified_log_file
+
+def delete_old_log_files(dir_path, uploaded_file_dates):
+    log_files_paths = get_file_paths(dir_path)
+    files_for_days_that_upload_failed = {}
+
+    for file_path in log_files_paths:
+        file_date_match = re.search(date_pattern, file_path)
+        file_date = file_date_match.group()
+
+        # Delete files for days that have a file uploaded in g-cloud
+        if file_date in uploaded_file_dates:
+            log.info(f"Deleting {file_path} because files for {file_date} already uploaded to drive")
+            os.remove(os.path.join(dir_path, file_path))
+
+        # Create a list of files for days that latest modified file failed to upload
+        else:
+            if file_date in files_for_days_that_upload_failed:
+                files_for_days_that_upload_failed[file_date].append(file_path)
+            else:
+                files_for_days_that_upload_failed[file_date] = []
+                files_for_days_that_upload_failed[file_date].append(file_path)
```

Comment:
This pattern is slightly cleaner if you use:
```python
if x not in dict:
   dict[x] = []
dict[x].append(y)
```

Reply:
updated thanks